src/backend/distributed/executor/adaptive_executor_repartition.c

```diff
@@ -0,0 +1,519 @@
+/*-------------------------------------------------------------------------
+ *
+ * adaptive_executor_repartition.c
```

Comment:
I think the adaptive executor is a bit of an implementation detail since you're only interacting with it via ExecuteTaskList (which originally comes from the router executor) and are also sending commands outside of the adaptive executor. 

Is there a better way to describe what we're doing here? Is it even specific to re-partitioning?

Reply:
umm currently this file is kind of specific to repartitioning because it has some repartition specific things such as generating FETCH_TASK query string from MAP_TASK( however currently this could well be done in the planner step, it is currently left empty because we dont know in which placement the map task succeeded in case of replication factor > 1, however we dont even use adaptive executor for repartition joins if that is the case.) 

There are other things such as creating schemas etc which are also repartition specific, so I guess in general I would say this file is repartition specific. So I will try to give a better description here

Reply:
I'm wondering whether we can split up the file into "directed acyclic graph execution logic" and "re-partitioning logic". Also, I think we probably do not need really call it "adaptive" because that's an implementation detail and soon-ish it will be the only way to execute re-partition joins.

Reply:
Makes sense.